### ISSUE TITLE
Error Stair Type Add

### ISSUE THREAD

**Issue by S59-Koara** [IfcOpenShell/IfcOpenShell#5261]
I seem to get an error when I try to add IfcStair Type. The following is the Log. There is an error from a new blank IFC4x3, but I will add the steps if necessary.

```
os: Windows
os_version: 10.0.22631
python_version: 3.11.7
architecture: ('64bit', 'WindowsPE')
machine: AMD64
processor: AMD64 Family 25 Model 33 Stepping 2, AuthenticAMD
blender_version: 4.2.1 LTS
bonsai_version: 0.7.11-alpha240830
bonsai_commit_hash: b4b2b60
last_actions: 
# ifcopenshell.api: aggregate.assign_object
>>> {"products": [{"cast_type": "entity_instance", "value": 42, "Name": "My Storey"}], "relating_object": {"cast_type": "entity_instance", "value": 36, "Name": "My Building"}}
# ifcopenshell.api: spatial.unassign_container
>>> {"products": "[]"}
# ifcopenshell.api: owner.create_owner_history
>>> {}
# ifcopenshell.api: geometry.edit_object_placement
>>> {"product": {"cast_type": "entity_instance", "value": 42, "Name": "My Storey"}, "matrix": {"cast_type": "ndarray", "value": [[1.0, 0.0, 0.0, 0.0], [0.0, 1.0, 0.0, 0.0], [0.0, 0.0, 1.0, 0.0], [0.0, 0.0, 0.0, 1.0]]}, "is_si": "False"}
# ifcopenshell.api: owner.update_owner_history
>>> {"element": {"cast_type": "entity_instance", "value": 42, "Name": "My Storey"}}
# operator: BIM_OT_enable_add_type
# operator: BIM_OT_add_type
# ifcopenshell.api: root.create_entity
>>> {"ifc_class": "IfcStairType", "predefined_type": "QUARTER_TURN_STAIR", "name": "TYPEX"}
# ifcopenshell.api: owner.create_owner_history
>>> {}
# operator: BIM_OT_add_stair
last_error: Traceback (most recent call last):
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\bonsai\bim\ifc.py", line 408, in execute_ifc_operator
    result = getattr(operator, "_execute")(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\type\operator.py", line 448, in _execute
    bpy.ops.bim.add_stair()
  File "C:\Program Files (x86)\Steam\steamapps\common\Blender\4.2\scripts\modules\bpy\ops.py", line 109, in __call__
    ret = _op_call(self.idname_py(), kw)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: Error: Python: Traceback (most recent call last):
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\bonsai\bim\ifc.py", line 408, in execute_ifc_operator
    result = getattr(operator, "_execute")(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\model\stair.py", line 206, in _execute
    props = obj.BIMStairProperties
            ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'BIMStairProperties'
Location: C:\Program Files (x86)\Steam\steamapps\common\Blender\4.2\scripts\modules\bpy\ops.py:109


ifc: Unsaved
schema: IFC4X3
preprocessor_version: IfcOpenShell 0.7.11-alpha240830
originating_system: IfcOpenShell 0.7.11-alpha240830
```

**Comment by Andrej730** [comment]
Fixed, thank you for the bug report!

**Comment by S59-Koara** [comment]
I tried to add IfcStairType in Bonsai alpha240902 and got an error.


![image](https://github.com/user-attachments/assets/91f990a0-2575-4977-8063-d5962fdfecde)
![image](https://github.com/user-attachments/assets/5dc30d85-2e6f-4350-99db-c349a184be06)


RuntimeError: Unable to find keyword in schema: STRAIGHT

The IfcStairTypeEnum only has STRAIGHT_RUN_STAIR or TWO_STRAIGHT_RUN_STAIR, but is it a problem to specify STRAIGHT?

https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcStairTypeEnum.htm

```
os: Windows
os_version: 10.0.22631
python_version: 3.11.7
architecture: ('64bit', 'WindowsPE')
machine: AMD64
processor: AMD64 Family 25 Model 33 Stepping 2, AuthenticAMD
blender_version: 4.2.1 LTS
bonsai_version: 0.8.1-alpha240902
bonsai_commit_hash: 1a8be00
last_actions: 
# ifcopenshell.api: geometry.edit_object_placement
>>> {"product": {"cast_type": "entity_instance", "value": 30, "Name": "My Site"}, "matrix": {"cast_type": "ndarray", "value": [[1.0, 0.0, 0.0, 0.0], [0.0, 1.0, 0.0, 0.0], [0.0, 0.0, 1.0, 0.0], [0.0, 0.0, 0.0, 1.0]]}, "is_si": "False"}
# ifcopenshell.api: owner.update_owner_history
>>> {"element": {"cast_type": "entity_instance", "value": 30, "Name": "My Site"}}
# ifcopenshell.api: aggregate.assign_object
>>> {"products": [{"cast_type": "entity_instance", "value": 36, "Name": "My Building"}], "relating_object": {"cast_type": "entity_instance", "value": 30, "Name": "My Site"}}
# ifcopenshell.api: spatial.unassign_container
>>> {"products": "[]"}
# ifcopenshell.api: owner.create_owner_history
>>> {}
# ifcopenshell.api: geometry.edit_object_placement
>>> {"product": {"cast_type": "entity_instance", "value": 36, "Name": "My Building"}, "matrix": {"cast_type": "ndarray", "value": [[1.0, 0.0, 0.0, 0.0], [0.0, 1.0, 0.0, 0.0], [0.0, 0.0, 1.0, 0.0], [0.0, 0.0, 0.0, 1.0]]}, "is_si": "False"}
# ifcopenshell.api: owner.update_owner_history
>>> {"element": {"cast_type": "entity_instance", "value": 36, "Name": "My Building"}}
# ifcopenshell.api: aggregate.assign_object
>>> {"products": [{"cast_type": "entity_instance", "value": 42, "Name": "My Storey"}], "relating_object": {"cast_type": "entity_instance", "value": 36, "Name": "My Building"}}
# ifcopenshell.api: spatial.unassign_container
>>> {"products": "[]"}
# ifcopenshell.api: owner.create_owner_history
>>> {}
# ifcopenshell.api: geometry.edit_object_placement
>>> {"product": {"cast_type": "entity_instance", "value": 42, "Name": "My Storey"}, "matrix": {"cast_type": "ndarray", "value": [[1.0, 0.0, 0.0, 0.0], [0.0, 1.0, 0.0, 0.0], [0.0, 0.0, 1.0, 0.0], [0.0, 0.0, 0.0, 1.0]]}, "is_si": "False"}
# ifcopenshell.api: owner.update_owner_history
>>> {"element": {"cast_type": "entity_instance", "value": 42, "Name": "My Storey"}}
# operator: BIM_OT_enable_add_type
# operator: BIM_OT_add_type
# ifcopenshell.api: root.create_entity
>>> {"ifc_class": "IfcStairType", "predefined_type": "QUARTER_TURN_STAIR", "name": "TYPEX"}
# ifcopenshell.api: owner.create_owner_history
>>> {}
# operator: BIM_OT_add_stair
# ifcopenshell.api: pset.add_pset
>>> {"product": {"cast_type": "entity_instance", "value": 66, "Name": "TYPEX"}, "name": "BBIM_Stair"}
# ifcopenshell.api: owner.create_owner_history
>>> {}
# ifcopenshell.api: pset.edit_pset
>>> {"pset": {"cast_type": "entity_instance", "value": 67, "Name": "BBIM_Stair"}, "properties": "{'Data': IfcText('{\"stair_type\": \"CONCRETE\", \"width\": 1.2000000476837158, \"height\": 1.0, \"number_of_treads\": 6, \"tread_run\": 0.30000001192092896, \"nosing_length\": 0.0, \"nosing_depth\": 0.0, \"base_slab_depth\": 0.25, \"top_slab_depth\": 0.25, \"has_top_nib\": true, \"tread_depth\": 0.25, \"custom_first_last_tread_run\": [0.0, 0.0]}')}"}
last_error: Traceback (most recent call last):
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\bonsai\bim\ifc.py", line 408, in execute_ifc_operator
    result = getattr(operator, "_execute")(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\type\operator.py", line 448, in _execute
    bpy.ops.bim.add_stair()
  File "C:\Program Files (x86)\Steam\steamapps\common\Blender\4.2\scripts\modules\bpy\ops.py", line 109, in __call__
    ret = _op_call(self.idname_py(), kw)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: Error: Python: Traceback (most recent call last):
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\bonsai\bim\ifc.py", line 408, in execute_ifc_operator
    result = getattr(operator, "_execute")(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\model\stair.py", line 221, in _execute
    update_ifc_stair_props(obj)
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\model\stair.py", line 97, in update_ifc_stair_props
    element.PredefinedType = "STRAIGHT"
    ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\ifcopenshell\entity_instance.py", line 325, in __setattr__
    self[index] = value
    ~~~~^^^^^^^
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\ifcopenshell\entity_instance.py", line 361, in __setitem__
    self.method_list[idx](self.wrapped_data, idx, entity_instance.unwrap_value(value))
  File "C:\Users\Owner\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\ifcopenshell\ifcopenshell_wrapper.py", line 8676, in setArgumentAsString
    return _ifcopenshell_wrapper.entity_instance_setArgumentAsString(self, i, a)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: Unable to find keyword in schema: STRAIGHT
Location: C:\Program Files (x86)\Steam\steamapps\common\Blender\4.2\scripts\modules\bpy\ops.py:109


ifc: Unsaved
schema: IFC4X3
preprocessor_version: IfcOpenShell 0.8.1-alpha240902
originating_system: IfcOpenShell 0.8.1-alpha240902
```

**Comment by Andrej730** [comment]
@S59-Koara it wasn't allowing other classes besides IfcStairFlight, fixed in the latest build after 07be50b
